client/src/main/java/com/netflix/conductor/client/task/WorkflowTaskCoordinator.java

```diff
@@ -23,426 +23,420 @@
 import com.netflix.conductor.common.metadata.tasks.TaskResult;
 import com.netflix.conductor.common.utils.RetryUtil;
 import com.netflix.discovery.EurekaClient;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
 import java.io.PrintWriter;
 import java.io.StringWriter;
 import java.util.Arrays;
-import java.util.Collections;
 import java.util.LinkedList;
 import java.util.List;
+import java.util.Objects;
 import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
-import java.util.concurrent.LinkedBlockingQueue;
-import java.util.concurrent.RejectedExecutionException;
 import java.util.concurrent.ScheduledExecutorService;
-import java.util.concurrent.ThreadPoolExecutor;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicInteger;
-
-import static com.netflix.conductor.client.task.WorkflowTaskMetrics.getPollTimer;
-import static com.netflix.conductor.client.task.WorkflowTaskMetrics.incrementTaskPollCount;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.concurrent.BasicThreadFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Manages the Task workers thread pool and server communication (poll, task update and acknowledgement).
- *
- * @author Viren
  */
 public class WorkflowTaskCoordinator {
 
-	private static final Logger logger = LoggerFactory.getLogger(WorkflowTaskCoordinator.class);
-
-	private TaskClient taskClient;
+    private static final Logger LOGGER = LoggerFactory.getLogger(WorkflowTaskCoordinator.class);
```

Comment:
There is a lot happening in this class, like WorkflowTaskExecution setup and the orchestration of polling for new messages and processing them. Think to break this into smaller composable classes. It would reduce complexity of the class.

Reply:
Refactored into `TaskRunnerConfigurer` and `TaskPollExecutor`